kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java

```diff
@@ -513,9 +516,15 @@ public static TaskSummaryList convertToTaskSummaryList(Collection<TaskSummary> t
                 .processInstanceId(taskSummary.getProcessInstanceId())
                 .status(taskSummary.getStatusId())
                 .skipable(taskSummary.isSkipable())
-                .subject(taskSummary.getSubject())
-                .build();
-        return task;
+                .subject(taskSummary.getSubject());
+         // alternative to that instanceof is to use default interface methods in task summary, but this requires java8 >
+         // in all implementors
+         if (taskSummary instanceof TaskSummaryImpl)   
+         {
+             builder.correlationKey(((TaskSummaryImpl)taskSummary).getCorrelationKey());
```

Comment:
promote to api.

Reply:
if we promote to API, we are breaking backward compatibility, I do not think that will be wise, taking into account the API is defined in a separate repository, which potentially have different implementations in other repositories different than JBPM

Reply:
There is no other implementation rather than JBPM one, so the interface can be changed